src/vstart.sh

```diff
@@ -1304,6 +1304,11 @@ mds_debug_scatterstat = true
 mds_verify_scatter = true
 EOF
     fi
+    if [ "$cephadm" -gt 0 ]; then
+        debug echo Setting mon public_network ...
+        public_network=$(ip route list | grep -w "$IP" | awk '{print $1}')
```

Comment:
This wont cover the ipv6 case, so will need to address that. my --ipv6 vstart PR still isn't landed so will go deal with it there :) 

I'll go make a comment on it so I don't forget.